src/lab.c

```diff
@@ -1368,12 +1368,24 @@ void CPUThink(GOBJ *event, GOBJ *hmn, GOBJ *cpu)
             goto CPULOGIC_RECOVER;
         }
 
+        // The default held button is PAD_BUTTON_A
+        int input = PAD_BUTTON_A;
+
+        if (LabOptions_CPU[OPTCPU_GRABRELEASE].option_val == CPUGRABRELEASE_AIRBORN)
+        {
+            input =  HSD_BUTTON_X;
```

Comment:
I think you should be able to change this line to `cpu_data->input.held |= HSD_BUTTON_X;`. Then you can remove all the other added code in this function.

Reply:
It turns out that this check can simply be added at the end of the ``OPTCPU_MASH`` match, that's probably what you meant and I missunderstood...

```C
if (LabOptions_CPU[OPTCPU_GRABRELEASE].option_val == CPUGRABRELEASE_AIRBORN)
{
        cpu_data->cpu.held |= HSD_BUTTON_X;
 }
 ```